### PR TITLE
Terser

### PR DESCRIPTION
**Pull request by Stanzilla**

### CODE REVIEW

**Comment by Stanzilla** [comment]
@LabhanshAgrawal hey, what do you think about this? File size is a bit bigger because I only went with the default options for terser but otherwise looks cool?

File: webpack.config.js [hunk 1]
```diff
@@ -53,6 +54,10 @@ module.exports = [
         }
       ])
     ],
+    optimization: {
+      minimize: true,
+      minimizer: [new TerserPlugin()]
+    },
```
Comment (LabhanshAgrawal) [inline webpack.config.js]: No need to add it here, as it(webpack job) is not really bundling anything (null-loader), just used to copy assets.

**Comment by LabhanshAgrawal** [comment]
Looks good, Is this for #4115 ?
What's the advantage of using terser(after required config) over babel minify?

**Comment by Stanzilla** [comment]
Yes, part of it. Mostly minify is pretty dead at the moment https://github.com/babel/minify/issues/952

**Comment by LabhanshAgrawal** [comment]
Can you add a condition for checking the isProd variable, would be easier to debug during development

**Comment by LabhanshAgrawal** [comment]
Need to change the webpack config a little to use the type information. I can't add changes to the pr, so I've pushed it [here](https://github.com/LabhanshAgrawal/hyper/tree/terser), you can cherrypick the latest commit, if the change seems ok to you.
Other than the above, the rest looks cool, we can configure terser after this.

**Pull request merged**